internal/docker/docker.go

```
@@ -0,0 +1,177 @@
+	}
+	fmt.Printf("Pulled image %s successfully\n", imageName)
+
+	defer reader.Close()
+	// c.cli.ImagePull is asynchronous.
+	// The reader needs to be read completely for the pull operation to complete.
+	// If stdout is not required, consider using io.Discard instead of os.Stdout.
+	io.Copy(os.Stdout, reader)
+
+	env := []string{
+		fmt.Sprintf("K3S_KUBECONFIG_OUTPUT=%s", constants.KubeconfigDockerPath),
+		"K3S_KUBECONFIG_MODE=666",
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		log.Error("Error getting current working directory: %v", err)
+	}
+
+	mounts := []mount.Mount{
+		{
+			Type:   mount.TypeBind,
+			Source: pwd,
+			Target: "/output",
+		},
+		{
+			Type:   mount.TypeVolume,
+			Source: "k3s-server",
+			Target: "/var/lib/rancher/k3s",
+		},
+		{
+			Type:   mount.TypeBind,
+			Source: fmt.Sprintf("%s/laptop/k3s-bootstrap/colony.yaml", pwd),
+			Target: "/var/lib/rancher/k3s/server/manifests/colony.yaml",
+		},
+		{
+			Type:   mount.TypeTmpfs,
+			Target: "/run",
+		},
+		{
+			Type:   mount.TypeTmpfs,
+			Target: "/var/run",
+		},
+	}
+
+	resp, err := c.cli.ContainerCreate(ctx, &containerTypes.Config{
+		Image: imageName,
+		Env:   env,
+		Cmd: []string{
+			"server",
+			"--disable=traefik,servicelb",
+			"--tls-san=colony",
+			"--node-label=colony.konstruct.io/node-type=colony",
+		},
+	}, &containerTypes.HostConfig{
+		Privileged:  true,
+		NetworkMode: "host",
+		Mounts:      mounts,
+	}, nil, nil, constants.ColonyK3sContainerName)
+
+	if err != nil {
+		log.Error("Error creating container: %v", err)
+	}
+
+	fmt.Printf("Created container with ID %s\n", resp.ID)
+
+	if err := c.cli.ContainerStart(ctx, resp.ID, containerTypes.StartOptions{}); err != nil {
+		panic(err)
+	}
+
+	waitInterval := 2 * time.Second
+
+	fmt.Printf("Checking for file %s every %v...\n", fmt.Sprintf("./%s", constants.KubeconfigHostPath), waitInterval)
+	waitForKubeconfig(fmt.Sprintf("./%s", constants.KubeconfigHostPath), waitInterval)
```

This function should return an error that should be handled here. If not, it will simply error out but continue execution.